### PR TITLE
Introduce `PUT` endpoint for `/events` API

### PR DESCRIPTION
**Pull request by petrpalata**
- Updated validateSchema script to check schema structure (static only)
- Updated schema to include PUT /events endpoint and errors
- Added examples for request PUT /events request variants and all error types

### CODE REVIEW

**Review by srri — DISMISSED** [review]

File: schemas/fingerprint-server-api.yaml
Comment (JuroUhlar) [inline schemas/fingerprint-server-api.yaml:1843]: @petrpalata why is it `tags` when the JS agent property is `tag`? 
Can we unify it? 

https://dev.fingerprint.com/docs/js-agent#tag

File: schemas/fingerprint-server-api.yaml
Comment (JuroUhlar) [inline schemas/fingerprint-server-api.yaml:1888]: @petrpalata aren't you missing `StateNotReady` here? 
```suggestion
                * `StateNotReady ` - The event specified with request id is not ready for updates yet. Try again.
```

File: schemas/fingerprint-server-api.yaml
Comment (JuroUhlar) [inline schemas/fingerprint-server-api.yaml:1849]: @petrpalata 
It seems when I supply a wrong JSON with multiple keys, I get an invalid JSON will multiple errors in the response: 

image

Comment (JuroUhlar) [reply schemas/fingerprint-server-api.yaml:1849]: Also passing unexpected but valid JSON produces "invalid json" error message 

image

File: schemas/fingerprint-server-api.yaml
Comment (petrpalata) [reply schemas/fingerprint-server-api.yaml:1843]: Seems to be a miss in the spec, I thought it's called `tags` in the Server API response. We need to fix this in the response itself but can update it here in the meantime.

**Comment by JuroUhlar** [comment]
Switching PR back to draft until uncovered implementation issue issues are addressed.

File: schemas/fingerprint-server-api.yaml
Comment (srri) [reply schemas/fingerprint-server-api.yaml:1849]: changed this to something more generic "request body is not valid", as it doesn't conform.

**Comment by petrpalata** [comment]
Waiting for [this PR](https://github.com/fingerprintjs/koala/pull/6197) to be merged, otherwise should be ready.

File: bin/validateSchema.ts
Comment (JuroUhlar) [inline bin/validateSchema.ts:626]: Let's remove these comments and console logs now

File: schemas/fingerprint-server-api.yaml
Comment (JuroUhlar) [reply schemas/fingerprint-server-api.yaml:1852]: Nice, those issues have been resolved, but the message still says `must provide one of: linkedId, suspect, tags to be updated` even though the property is `tag` (not `tags`)

**Review by JuroUhlar — CHANGES_REQUESTED** [review]
Besides the two issues I pointed out in threads, it looks good to me. But I also committed a bunch into this branch so let's wait for @ilfa 's review before merging. Thanks!

File: schemas/fingerprint-server-api.yaml
Comment (petrpalata) [reply schemas/fingerprint-server-api.yaml:1852]: Let us fix that on the backend next week, I wouldn't be afraid to merge this though because it's a very minor thing.

**Review by JuroUhlar — DISMISSED** [review]

**Pull request merged**
